lib/main.js

```diff
@@ -283,29 +292,34 @@ class MyAlgoConnect {
 	async signTransaction(transaction) {
 		let txn;
 
+		if (this.currentSigntxPopup) {
+			if (this.currentSigntxPopup.closed) {
+				this.currentSigntxPopup = null;
+			}
+			else {
+				this.currentSigntxPopup.focus();
+				throw new Error(Errors.WINDOW_IS_OPENED);
+			}
+		}
+
 		if (Array.isArray(transaction))
 			txn = Array.from(transaction).map(tx => prepareTxn(tx));
 		else
 			txn = prepareTxn(transaction);
 
-		if (this.currentSigntxPopup) {
-			this.currentSigntxPopup.focus();
-			throw new Error(Errors.WINDOW_IS_OPENED);
-		}
-
 		try {
 			this.currentSigntxPopup = openPopup(this.url + "/signtx.html");
 
 			await this.waitForWindowToLoad(this.currentSigntxPopup);
 
 			// Send transaction info
 			const res = await this.bridge.sendMessage(
-				this.currentSigntxPopup, { method: "transaction", params: { txn } },
-				this.url, { waitForReply: true, timeout: this.timeout }
+				this.currentSigntxPopup,
+				{ method: "transaction", params: { txn } },
+				this.url, this.options
 			);
 
-			if (this.currentSigntxPopup)
-				this.currentSigntxPopup.close();
+			this.closeWindow(this.currentSigntxPopup);
 			this.currentSigntxPopup = null;
```

Comment:
maybe you can move this this.currentSigntxPopup = null; into closeWindow function